src/test/controlobjecttest.cpp

```diff
@@ -49,4 +49,16 @@ TEST_F(ControlObjectTest, aliasRetrieval) {
     EXPECT_EQ(ControlObject::getControl(ckAlias), co.get());
 }
 
+TEST_F(ControlObjectTest, persistence) {
```

Comment:
Thanks for the test! To emulate shutting down Mixxx you could destroy the user settings in between creations of the CO:

```c++
// create CO with persist and check its value
m_pConfig->save();
m_pConfig.reset(new UserSettings(getTestDataDir().filePath("test.cfg")));
ControlDoublePrivate::setUserConfig(m_pConfig);
// create new CO with persist and check its value
```

You could also test other cases (invalid value in the config) by setting it manually to an invalid or empty value:
```c++
m_pConfig->set(ck, "notanumber");
```